### ISSUE TITLE
NPE in ServerRequestGetRewards.onRequestSucceeded

### ISSUE THREAD

**Issue by Jeff11** [BranchMetrics/android-branch-deep-linking#284]
On version 2.0.2 we get many NPE in this method and the app crashes! 

Here's the stacktrace:

> Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'java.util.Iterator org.json.JSONObject.keys()' on a null object reference
       at io.branch.referral.ServerRequestGetRewards.onRequestSucceeded(ServerRequestGetRewards.java:46)
       at io.branch.referral.Branch$BranchPostTask.onPostExecute(Branch.java:2472)
       at io.branch.referral.Branch$BranchPostTask.onPostExecute(Branch.java:2320)
       at android.os.AsyncTask.finish(AsyncTask.java:651)
       at android.os.AsyncTask.access$500(AsyncTask.java:180)
       at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:668)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:158)
       at android.app.ActivityThread.main(ActivityThread.java:7229)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)

**Comment by sojanpr** [comment]
@Jeff11  Sorry that you are facing this issue. We are checking the issue and update you asap

**Comment by sojanpr** [comment]
@Jeff11  Thanks for notifying us the above issue. Our analysis showed that this issue is caused by a short outage happened on our API service during last week. We got recovered immediately but this issue seems like caused by the API requests happened during that time. 

We have opened a  new PR (https://github.com/BranchMetrics/android-branch-deep-linking/pull/285) to proactively fix if this happens in future. Will be added with our next release
